server/tso/allocator_manager.go

```diff
@@ -547,62 +598,80 @@ func (am *AllocatorManager) ClusterDCLocationChecker() {
 		return
 	}
 	am.mu.Lock()
-	defer am.mu.Unlock()
 	// Clean up the useless dc-locations
 	for dcLocation := range am.mu.clusterDCLocations {
 		if _, ok := newClusterDCLocations[dcLocation]; !ok {
 			delete(am.mu.clusterDCLocations, dcLocation)
 		}
 	}
+	// May be used to rollback the updating after
+	newDCLocations := make([]string, 0)
 	// Update the new dc-locations
 	for dcLocation, serverIDs := range newClusterDCLocations {
 		if _, ok := am.mu.clusterDCLocations[dcLocation]; !ok {
-			am.mu.clusterDCLocations[dcLocation] = &dcLocationInfo{
-				serverIDs: serverIDs,
-				suffix:    -1,
+			am.mu.clusterDCLocations[dcLocation] = &DCLocationInfo{
+				ServerIDs: serverIDs,
+				Suffix:    -1,
 			}
+			newDCLocations = append(newDCLocations, dcLocation)
 		}
 	}
 	// Only leader can write the TSO suffix to etcd in order to make it consistent in the cluster
 	if am.member.IsLeader() {
 		for dcLocation, info := range am.mu.clusterDCLocations {
-			if info.suffix > 0 {
+			if info.Suffix > 0 {
 				continue
 			}
 			suffix, err := am.getOrCreateLocalTSOSuffix(dcLocation)
 			if err != nil {
 				log.Warn("get or create the local tso suffix failed", zap.String("dc-location", dcLocation), errs.ZapError(err))
 				continue
 			}
-			am.mu.clusterDCLocations[dcLocation].suffix = suffix
+			if suffix > am.mu.maxSuffix {
+				// Because the number of suffix bits is changing dynamically according to the dc-location number,
+				// there is a corner case may cause the Local TSO is not unique while member changing.
+				// Example:
+				//     t1: xxxxxxxxxxxxxxx1 | 11
+				//     t2: xxxxxxxxxxxxxxx | 111
+				// Please take a look at https://github.com/tikv/pd/issues/3260 for more details.
+				if CalSuffixBits(suffix) != CalSuffixBits(am.mu.maxSuffix) {
+					info.NeedSyncMaxTSO = true
+				}
+				am.mu.maxSuffix = suffix
+			}
+			am.mu.clusterDCLocations[dcLocation].Suffix = suffix
+		}
+	} else {
+		// Follower should check and update the am.mu.maxSuffix
+		maxSuffix, err := am.getMaxLocalTSOSuffix()
+		if err != nil {
+			log.Error("get the max local tso suffix from etcd failed", errs.ZapError(err))
+			// Rollback the new dc-locations we update before
+			for _, dcLocation := range newDCLocations {
+				delete(am.mu.clusterDCLocations, dcLocation)
+			}
+		} else if maxSuffix > am.mu.maxSuffix {
+			am.mu.maxSuffix = maxSuffix
```

Comment:
Why `follower` need to maintain the `maxSuffix`?

Reply:
If the follower holds a Local TSO Allocator leader, it will be used to pass as an argument [here](https://github.com/tikv/pd/pull/3267/files#diff-864112911592e078c8ec9a72330c17cc7473a525e144f2bde0c2462a8f02d02dR99) while generating a Local TSO.